generator.yml

```diff
@@ -70,6 +70,18 @@ defconfigs:
   - &x86_64     {config: defconfig,                                                                << : *kernel_modules}
   - &x86_64_gki {config: gki_defconfig,                                                            << : *kernel_modules}
   - &x86_64_cut {config: x86_64_cuttlefish_defconfig,                                              << : *kernel_modules}
+allconfigs:
+  - &arm32_mn_am {config: allmodconfig, << : *arm-triple, << : *kernel_modules}
+  - &arm32_mn_an {config: allnoconfig, << : *arm-triple, << : *kernel_modules}
+  - &arm32_mn_ay {config: allyesconfig, << : *arm-triple, << : *kernel_modules}
+  #                                      https://github.com/ClangBuiltLinux/linux/issues/1212
+  - &arm64_mn_am {config: [allmodconfig, CONFIG_UBSAN_UNSIGNED_OVERFLOW=n], << : *arm64-triple, << : *kernel_modules}
+  - &arm64_mn_an {config: allnoconfig, << : *arm64-triple, << : *kernel_modules}
+  - &arm64_mn_ay {config: [allyesconfig, CONFIG_UBSAN_UNSIGNED_OVERFLOW=n], << : *arm64-triple, << : *kernel_modules}
+  #                                       https://github.com/ClangBuiltLinux/linux/issues/256
+  - &x86_64_mn_am {config: [allmodconfig, CONFIG_UBSAN_UNSIGNED_OVERFLOW=n], << : *kernel_modules}
+  - &x86_64_mn_an {config: allnoconfig, << : *kernel_modules}
+  - &x86_64_mn_ay {config: [allyesconfig, CONFIG_UBSAN_UNSIGNED_OVERFLOW=n], << : *kernel_modules}
```

Comment:
These don't need to be in their own `allconfigs` section; can you put them in `defconfigs` (and retitle `defconfigs` to `configs` if it's more appropriate?

I'm not sure what `mn` is ("mainline").  Maybe `arm32_allyes`, `arm32_allno`, `arm32_allmod` is slightly more readable, if more verbose.

Noted on a later commit, but revisited this earlier commit:

I really really don't like the idea of conflating a config with a branch of a tree.  I'd rather have `all{no|yes|mod}config`s for each arch, not dependent on a tree.  I get this is trying to be precise due to https://github.com/ClangBuiltLinux/linux/issues/256, but my recommendation is to just disable `CONFIG_UBSAN_UNSIGNED_OVERFLOW` for `all{yes|mod}` for now, or come up with a way to factor out the fragment, such that the list of `defconfigs` is purely a config name (with optional fragment) and target triple.

Reply:
> These don't need to be in their own `allconfigs` section; can you put them in `defconfigs` (and retitle `defconfigs` to `configs` if it's more appropriate?

Sure, I can do that.

> I'm not sure what `mn` is ("mainline").  Maybe `arm32_allyes`, `arm32_allno`, `arm32_allmod` is slightly more readable, if more verbose.

Yes, mn is mainline, nx is next, and 510 is 5.10. I can make them more verbose.

> Noted on a later commit, but revisited this earlier commit:
> 
> 
> 
> I really really don't like the idea of conflating a config with a branch of a tree.  I'd rather have `all{no|yes|mod}config`s for each arch, not dependent on a tree.  I get this is trying to be precise due to https://github.com/ClangBuiltLinux/linux/issues/256, but my recommendation is to just disable `CONFIG_UBSAN_UNSIGNED_OVERFLOW` for `all{yes|mod}` for now, or come up with a way to factor out the fragment, such that the list of `defconfigs` is purely a config name (with optional fragment) and target triple.

That is fine by me. I just worry that one tree's target is going to break so then we disable the config for all of the trees and lose coverage. But at the same time, your suggestion is way simpler so we can go with that.

Realistically, we can solve that concern by having the ability to apply mbox files on top of the tree to work around issues, which is an outstanding tuxsuite request.

Reply:
Yes, the mbox file approach SGTM.